### PR TITLE
Refactors deathsounds, adds 2 new ones (lizards and borgs)

### PR DESCRIPTION
**Pull request by PKPenguin321**
🆑
code: Xeno deathsounds are less snowflakey. Deathsounds are now easy to add to mobs in code.
soundadd: Lizards and borgs now have sounds that play on *deathgasp, like xenos. So as to not nerf silent kills, anything that mutes (gags, poison, being a mime) or having 50 or more oxygen damage will prevent you from making a sound when you deathgasp.
/:cl:

- deathsounds are sounds played when *deathgasp is used, either manually or automatically
- deathsounds are a var on /mob
- made xenos use the var
- removed the hack for xenos deathsound in the deathgasp emote and made it use the var
- 2 new deathsounds: one for lizards, one for borgs

**ALL SOUNDS ARE SUBJECT TO CHANGE.** Please give feedback or suggestions if you have an improvement or better sounds. Info on the lizard sound can be found in the credits.txt I added, and the borg deathsound is a completely unused one from old TF2 builds that I can't find a license to (I'm not sure it even has a license on it). I'd like to maybe add human deathsounds too, but I can't find any sounds I like enough (and I would want to add male/female variants).

[Here is a video with sound to demonstrate the new sounds.](https://file.house/tBRs.mp4) Outdated now, see the new borg deathsound linked in comments below.

- [x] Lizard deathsound isn't applied to people who pick lizard race roundstart - needs to be fixed before merge
- [x] Make the sound not play if you're mute (mimes or poison or otherwise), maybe make it not play if you have too much oxy damage so you can do silent chokeouts/leave people in crit for them to die silently

### CODE REVIEW

File: code/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -997,6 +997,7 @@
 
 /mob/living/carbon/human/species/lizard
 	race = /datum/species/lizard
+	deathsound = 'sound/voice/lizard/deathsound.ogg'
```
Comment (Poojawa) [inline code/modules/mob/living/carbon/human/human.dm]: Make this sort of thing an On species gain/lose sort of thing. That'll sort out all of the sounds

Comment (PKPenguin321) [reply code/modules/mob/living/carbon/human/human.dm]: I ended up figuring this out and fixing it up on my own before seeing this, but thank you for your help regardless!

Comment (Poojawa) [reply code/modules/mob/living/carbon/human/human.dm]: Yee yee. Does it still apply the correct sound if you swap species?

File: code/modules/mob/living/carbon/human/species.dm [hunk 2]
```diff
@@ -44,6 +44,7 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 	var/siemens_coeff = 1 //base electrocution coefficient
 	var/damage_overlay_type = "human" //what kind of damage overlays (if any) appear on our species when wounded?
 	var/fixed_mut_color = "" //to use MUTCOLOR with a fixed color that's independent of dna.feature["mcolor"]
+	var/deathsound//used to set the mobs deathsound on species change
```
Comment (ShizCalev) [inline code/modules/mob/living/carbon/human/species.dm]: put a space before the // please.

File: code/modules/mob/living/emote.dm [hunk 3]
```diff
@@ -87,8 +87,12 @@
...
 	. = ..()
 	message_simple = initial(message_simple)
-	if(. && isalienadult(user))
-		playsound(user.loc, 'sound/voice/hiss6.ogg', 80, 1, 1)
+	if(. && user.deathsound)
+		if(isliving(user))
+			var/mob/living/L = user
+			if(!L.can_speak_vocal() || L.oxyloss >= 50)
+				return//stop the message if: oxyloss too high/cant speak
+		playsound(user.loc, user.deathsound, 80, 1, 1)
```
Comment (ShizCalev) [inline code/modules/mob/living/emote.dm]: drop the .loc

**Comment by ATH1909** [comment]
Give bluspace golems the enderman death noise/sound as their death sound.

**Comment by anconfuzedrock** [comment]
Moth deathscream?!?!?!

**Comment by Dorsisdwarf** [comment]
borg *deathgasp sound should probbably be more in line with their standard harsh electric *buzz *beep *buzz2 etc

**Comment by PKPenguin321** [comment]
@Dorsisdwarf i liked your idea, tried my hand at it. let me know what you think
**[new borg deathsound link](https://file.house/AGb-.mp3)**

**Comment by actioninja** [comment]
I could record myself screaming like I'm dying if you want.

**Comment by PKPenguin321** [comment]
If you'd like to and think you can make a sound that's clean enough to put in the game and that players will like, go ahead

**Comment by anconfuzedrock** [comment]
the new sound is loud, it should be like when they start up, but "ow"

**Review by AnturK — DISMISSED** [review]
simple animals will now have both `deathsound` and `death_sound`. Unify these.

**Comment by PKPenguin321** [comment]
Good point, will do

**Comment by PKPenguin321** [comment]
@aconfuzedrock
Which sound are you referring to?

**Comment by PKPenguin321** [comment]
@ anconfuzedrock
Which sound are you referring to?

**Comment by anconfuzedrock** [comment]
I honestly have no idea pk is a joke

**Comment by improvedname** [comment]
Now time for human deathsound

**Comment by improvedname** [comment]
Its only fair why should lizard/borgs get death sound alarms to nearby people when the glorious master race humans. Cant have roblox oof

File: code/modules/mob/living/simple_animal/hostile/megafauna/drake.dm [hunk 4]
```diff
@@ -60,11 +60,11 @@ Difficulty: Medium
 	medal_type = BOSS_MEDAL_DRAKE
 	score_type = DRAKE_SCORE
 	deathmessage = "collapses into a pile of bones, its flesh sloughing away."
-	death_sound = 'sound/magic/demon_dies.ogg'
+	deathsound = 'sound/magic/demon_dies.ogg'
 	var/datum/action/small_sprite/smallsprite = new/datum/action/small_sprite/drake()
-	
+
```
Comment (PKPenguin321) [inline code/modules/mob/living/simple_animal/hostile/megafauna/drake.dm]: accidentally deleted a bunch of whitespace in this file, can undo if maintainers want

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**